proving_system/stark/src/air/frame.rs

```diff
@@ -52,6 +52,25 @@ impl<F: IsField> Frame<F> {
         Self::new(data, 1)
     }
 
+    pub fn get_trace_evaluations(
+        trace_polys: &[Polynomial<FieldElement<F>>],
+        x: &FieldElement<F>,
+        frame_offsets: &[usize],
+        primitive_root: &FieldElement<F>,
+    ) -> Vec<Vec<FieldElement<F>>> {
+        let mut evaluations = Vec::with_capacity(trace_polys.len());
+        let evaluation_points: Vec<FieldElement<F>> = frame_offsets
+            .iter()
+            .map(|offset| x * primitive_root.pow(*offset))
+            .collect();
+
+        for poly in trace_polys {
+            evaluations.push(poly.evaluate_slice(&evaluation_points));
+        }
```

Comment:
you could have the same programming style than in the precedent assignment

Reply:
done!